examples/07_advanced/plot_haxby_mass_univariate.py

```diff
@@ -34,134 +34,166 @@
+print(f"Mask nifti image (3D) is located at: {haxby_dataset.mask}")
+print(f"Functional nifti image (4D) is located at: {haxby_dataset.func[0]}")
 
 ##############################################################################
 # Restrict to faces and houses
 import numpy as np
 import pandas as pd
+
 labels = pd.read_csv(haxby_dataset.session_target[0], sep=" ")
-conditions = labels['labels']
+conditions = labels["labels"]
 categories = conditions.unique()
 conditions_encoded = np.zeros_like(conditions)
 for c, category in enumerate(categories):
     conditions_encoded[conditions == category] = c
-sessions = labels['chunks']
-condition_mask = conditions.isin(['face', 'house'])
+sessions = labels["chunks"]
+condition_mask = conditions.isin(["face", "house"])
 conditions_encoded = conditions_encoded[condition_mask]
 
 ##############################################################################
 # Mask data
 mask_filename = haxby_dataset.mask
 from nilearn.image import index_img
 from nilearn.maskers import NiftiMasker
+
 nifti_masker = NiftiMasker(
     smoothing_fwhm=8,
     mask_img=mask_filename,
-    memory='nilearn_cache', memory_level=1)  # cache options
+    memory="nilearn_cache",
+    memory_level=1,
+)  # cache options
 func_filename = haxby_dataset.func[0]
-func_reduced = index_img(func_filename,
-                         condition_mask)
+func_reduced = index_img(func_filename, condition_mask)
 fmri_masked = nifti_masker.fit_transform(func_reduced)
 
 # We consider the mean image per session and per condition.
 # Otherwise, the observations cannot be exchanged at random because
 # a time dependence exists between observations within a same session.
 n_sessions = np.unique(sessions).size
-grouped_fmri_masked = np.empty((2 * n_sessions,  # two conditions per session
-                                fmri_masked.shape[1]))
+grouped_fmri_masked = np.empty(
+    (2 * n_sessions, fmri_masked.shape[1])  # two conditions per session
+)
 grouped_conditions_encoded = np.empty((2 * n_sessions, 1))
 
 for s in range(n_sessions):
     session_mask = sessions[condition_mask] == s
-    session_house_mask = np.logical_and(session_mask,
-                                        conditions[condition_mask] == 'house')
-    session_face_mask = np.logical_and(session_mask,
-                                       conditions[condition_mask] == 'face')
+    session_house_mask = np.logical_and(
+        session_mask, conditions[condition_mask] == "house"
+    )
+    session_face_mask = np.logical_and(
+        session_mask, conditions[condition_mask] == "face"
+    )
     grouped_fmri_masked[2 * s] = fmri_masked[session_house_mask].mean(0)
     grouped_fmri_masked[2 * s + 1] = fmri_masked[session_face_mask].mean(0)
-    grouped_conditions_encoded[2 * s] = conditions_encoded[
-        session_house_mask][0]
+    grouped_conditions_encoded[2 * s] = conditions_encoded[session_house_mask][
+        0
```

Comment:
:-(

Reply:
That's sad, but I don't think it's much worse than what we had before haha